pkg/reconciler/openshift/tektonaddon/transformer.go

```diff
@@ -133,3 +137,79 @@ func setVersionedNames(operatorVersion string) mf.Transformer {
 		return nil
 	}
 }
+
+// replacePACTriggerTemplateImages replaces images in all the TriggerTemplates that
+// PAC creates. These TriggerTemplates have the "same" PAC image in all
+// TaskRun.Spec.TaskSpec.Steps irrespective of the step (or container) name; this
+// is what makes this transformer a bit different that other replace-image
+// transformers. In other transformers, the images are replaced based on the
+// container name in Deployments, but here the container names in steps are
+// irrelevant because we need to replace any image in "all" steps present with the
+// PAC image.
+func replacePACTriggerTemplateImages(pacImage string) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() != "TriggerTemplate" {
+			return nil
+		}
+
+		tt := &triggersv1beta1.TriggerTemplate{}
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, tt)
+		if err != nil {
+			return err
+		}
+
+		for i, resourceTemplate := range tt.Spec.ResourceTemplates {
+			// let' start by checking that we are only dealing with a TaskRun
+
+			// first we need to decode the raw ResourceTemplate
+			rawReader := mf.Reader(bytes.NewReader(resourceTemplate.RawExtension.Raw))
+			rawObjects, err := rawReader.Parse()
+			if err != nil {
+				return err
+			}
+
+			if len(rawObjects) == 0 {
+				// if no object is present in the ResourceTemplate, we don't want to proceed
+				return nil
+			} else if len(rawObjects) > 1 {
+				// while this will never happen, let's check for it just in case
+				return fmt.Errorf("more than one resources present in a ResourceTemplate")
+			}
+
+			// we only care about TaskRuns in ResourceTemplate for image replacement
+			if rawObjects[0].GetKind() != "TaskRun" {
+				return nil
+			}
+
+			// now that we know it's a TaskRun, let's decode and move forward
+			taskRun := pipelinev1beta1.TaskRun{}
+			decoder := json.NewDecoder(bytes.NewBuffer(resourceTemplate.RawExtension.Raw))
+			if err := decoder.Decode(&taskRun); err != nil {
+				return err
+			}
+
+			for j, step := range taskRun.Spec.TaskSpec.Steps {
```

Comment:
this method with replace any image in taskruns of triggertemplate, we should fins a pattern with name or something that pac image is replaced, this may result in replacing images of any other triggertemplates also

Reply:
I thought about this as well, but there is no pattern here. I'll ping @chmouel and @sm43 if PAC can guarantee same step names in the future.

Reply:
what kind of guarantee you need? i am not quite understanding the question,

Reply:
@chmouel any pattern that can we identify to replace the image, like step name, etc.
For now, I'll move ahead with what @piyush-garg has suggested.

Reply:
or it could be any label in the triggerTemplate which this function could use to ensure that the triggertemplate it is working on is from PAC. 

cc @piyush-garg @concaf may be we can keep PAC project as it is and add label through [this](https://github.com/tektoncd/operator/blob/42d68c8428211e993b8455d7de292e0773fc54c9/hack/fetch-releases.sh#L97) if needed.

Reply:
@nikhil-thomas for now, since we don't have any other TriggerTemplate transformer, the current solution should work but going forward we could do something like that, WDYT?

@piyush-garg I've made the changes at 668b77014b8107e0f24ccdda5c25ddbbefb8be71

Reply:
@nikhil-thomas thinking about it, what you said makes sense 😅 added a check here - f43d62ad909f6dc783c22631b98dbaeb9022e465